### PR TITLE
fix(ios): add xcode 14 workaround (turn off signing resource bundles) for pods

### PR DESCRIPTION
**Pull request by kelset**
## Summary

This is inspired by the Expo workaround https://github.com/expo/expo/commit/d970a9ecbb15b554b6fa23e8d43006fd15152028 to address an issue that cocoapods has with Xcode 14: https://github.com/CocoaPods/CocoaPods/issues/11402

This wants to address this https://github.com/facebook/react-native/issues/34673 in a way that we can also cherry-pick on the 0.70 branch.

## Changelog

[iOS] [Fixed] - add xcode 14 workaround (turn off signing resource bundles) for `React-Core`

## Test Plan

Tested locally by opening RNTester via Xcode 14.0.1, and targetting my iPhone as device. After applying the patch, the error for React Core AccessibilityResources disappears.

Also, added ruby test for new patch.

### CODE REVIEW

**Review by cipolleschi — CHANGES_REQUESTED** [review]
We should refactor this in the `ReactNativePodsUtils` file and add some tests.

**Comment by danilobuerger** [comment]
I don't like how this modifies all pods, not only the ones included by rn. We only need this for one resource bundle: 

https://github.com/facebook/react-native/blob/88b7b640a74bafd918b8b1cd5d58e1f5ddfb730a/React-Core.podspec#L46

**Comment by Kudo** [comment]
> I don't like how this modifies all pods, not only the ones included by rn. We only need this for one resource bundle:

we use this strategy because we also have other resource bundles at expo 😅 please do whatever best fit for react-native templates.

**Comment by kelset** [comment]
yeah I agree that this workaround is not ideal - my current concern is that if we do it for just that specific resource bundle, developers might still hit the issue. I feel that maybe we should just add this to the template so that developers will have to "opt in" into copying this pod postinstall over in their setups understanding what this is doing.

**Comment by kelset** [comment]
also, it seems that by going for this route (of not doing the signing) you might incurr in Apple rejection along the lines of:

>  The binary with bundle identifier 'SOME_IDENTIFIER' at path [SOME_APP.app] contains an invalid signature. Make sure you have signed your application with a distribution certificate, not an ad hoc certificate or a development
certificate. Verify that the code signing settings in Xcode are correct at the target level (which override
any values at the project level). If you are certain your code signing settings are correct, choose 'Clean
All' in Xcode, delete the 'build' directory in the Finder, and rebuild your release target.

see [more details here](https://github.com/CocoaPods/CocoaPods/issues/11402#issuecomment-1258608914).

So maybe we should go another route and tell folks to go for

```
 installer.generated_projects.each do |project|
    project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings["DEVELOPMENT_TEAM"] = "TEAM_ID"
        end
    end
  end
```

instead, as [suggested here](https://github.com/facebook/react-native/issues/34673#issuecomment-1248122805)? (meaning, configuring the team account to sign it all)

**Comment by danilobuerger** [comment]
I think the rejects are "unrelated" and not caused by unsigned resource bundles. We have gone through review a couple of times now with unsigned resource bundles and xcode 14. 🤷‍♂️

We could just add a new func like we did for the m1 so people can opt out more easily without removing the whole post install func call.

**Comment by cipolleschi** [comment]
What if we modify this to act only on the `React-Core` pod, like we did [here](https://github.com/facebook/react-native/blob/main/scripts/cocoapods/new_architecture.rb#L55)?

Also, it would be nice to wrap this code in a function that lives in the [`ReactNativePodsUtils`](https://github.com/facebook/react-native/blob/main/scripts/cocoapods/utils.rb#L9) class with some tests.

@kelset, if you are busy, I can take it. Let me know what do you prefer

File: scripts/cocoapods/__tests__/utils-test.rb [hunk 1]
```diff
@@ -382,6 +382,44 @@ def test_fixReactBridgingHeaderSearchPaths_correctlySetsTheHeaderSearchPathsForA
...
+        pods_projects_mock = PodsProjectMock.new([],
+            {
+            "hermes-engine" => {},
+            "React-Core" => [ react_core_debug_config, react_core_release_config ],
+        })
+
+        installer = InstallerMock.new(pods_projects_mock, [
+            AggregatedProjectMock.new(user_project_mock)
+        ])
+
```
Comment (cipolleschi) [inline scripts/cocoapods/__tests__/utils-test.rb:408]: I would probably add and assert similar to the one we have at line 416 to make sure that the `CODE_SIGNING_ALLOWED` flag is YES before the invocation of `turn_off_resource_bundle_react_core`. In this way the test properly assert that we update from `YES` to `NO`.

File: scripts/cocoapods/__tests__/utils-test.rb [hunk 2]
```diff
@@ -382,6 +382,44 @@ def test_fixReactBridgingHeaderSearchPaths_correctlySetsTheHeaderSearchPathsForA
...
+        user_project_mock = UserProjectMock.new("a/path", [
+            prepare_config("Debug"),
+            prepare_config("Release"),
+        ])
+
+        pods_projects_mock = PodsProjectMock.new([],
+            {
+            "hermes-engine" => {},
+            "React-Core" => [ react_core_debug_config, react_core_release_config ],
```
Comment (cipolleschi) [inline scripts/cocoapods/__tests__/utils-test.rb:402]: I would aslo add another target which is not React-Core and which has the BuildConfigurationMock with the `CODE_SIGNING_ALLOWED` set to YES. And we assert that it stays to YES. In this way we test that we are actually only changing the React-Core pod.

File: scripts/cocoapods/__tests__/utils-test.rb [hunk 3]
```diff
@@ -382,6 +382,44 @@ def test_fixReactBridgingHeaderSearchPaths_correctlySetsTheHeaderSearchPathsForA
...
+        installer = InstallerMock.new(pods_projects_mock, [
+            AggregatedProjectMock.new(user_project_mock)
+        ])
+
+        # Act
+        ReactNativePodsUtils.turn_off_resource_bundle_react_core(installer)
+
+        # Assert
+        # something like this?
+        # assert_equal(pods_projects_mock.build_settings['CODE_SIGNING_ALLOWED'], "NO")
```
Comment (cipolleschi) [inline scripts/cocoapods/__tests__/utils-test.rb:414]: Why do we have these comments? Something left to address here?

**Review by cipolleschi — CHANGES_REQUESTED** [review]
Hi @kelset! Thank you for doing this! I left some more improvements for the tests. Let me know what do you think!

**Comment by taylorkline** [comment]
> we should be able to remove this once CocoaPods catches up to it, see more details here https://github.com/CocoaPods/CocoaPods/issues/11402

Will be fixed in CocoaPods 1.12.0: https://github.com/CocoaPods/CocoaPods/issues/11402#issuecomment-1379702414

**Pull request closed without merging**